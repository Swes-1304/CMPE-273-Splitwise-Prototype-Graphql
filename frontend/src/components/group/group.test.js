import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Group from './group';
import Navheader from '../navbar/navbar';

const component = TestRenderer.create(
  <MemoryRouter>
    <Group>
      <Navheader />
    </Group>{' '}
  </MemoryRouter>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for leave grooup', async () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Group>
        <Navheader />
      </Group>{' '}
    </MemoryRouter>
  );
  expect(getByTestId('leavegroup')).toHaveTextContent('Leave Group');
});
