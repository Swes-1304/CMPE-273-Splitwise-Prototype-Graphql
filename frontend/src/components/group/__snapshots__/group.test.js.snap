// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div>
    <div>
      <nav
        className="navbar-default navbar navbar-expand navbar-light"
      >
        <a
          className="navbar-brand"
          classname="Navbar-Brand"
          href="/"
          variant="light"
        >
          <img
            alt="icon"
            className="icon"
            src="/Group_Photos/Splitwiseicon.png"
          />
          <span
            className="NavItem"
          >
            Splitwise
          </span>
        </a>
        <div
          aria-expanded={null}
          className="justify-content-end navbar-collapse collapse"
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <button
                  className="login-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    Login 
                  </a>
                </button>
                or
                 
                <button
                  className="Signup-default btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="grouppage-flex"
    >
      <div>
        <div
          className="sidebar"
        >
          <div
            className="flex-column nav"
            onKeyDown={[Function]}
          >
            <div
              className="sidebar-wrapper"
            >
              <ul
                className="navlist"
              >
                <li
                  className="nav-item"
                >
                  <li
                    className="nav-item1"
                  >
                    <a
                      aria-current={null}
                      className="nav-link1"
                      href="/dashboard"
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "inherit",
                          "textDecoration": "none",
                        }
                      }
                    />
                  </li>
                   
                  <hr />
                  <a
                    aria-current={null}
                    className="nav-link1"
                    href="/dashboard"
                    onClick={[Function]}
                  >
                    Dashboard
                  </a>
                </li>
                <hr />
                <li
                  className="nav-item1"
                >
                  <a
                    aria-current={null}
                    className="nav-link1"
                    href="/recentactivity"
                    onClick={[Function]}
                  >
                    Recent Acitivity
                  </a>
                </li>
                <hr />
                <li
                  className="nav-item1"
                >
                  <a
                    aria-current={null}
                    className="nav-link1"
                    href="/profile"
                    onClick={[Function]}
                  >
                    My Profile
                  </a>
                </li>
                <hr />
                <li
                  className="nav-item1"
                >
                  <a
                    aria-current={null}
                    className="nav-link1"
                    href="/mygroups"
                    onClick={[Function]}
                  >
                    My Groups
                  </a>
                </li>
                <li>
                  <a
                    aria-current={null}
                    className="nav-link2"
                    href="/createnewgroup"
                    onClick={[Function]}
                  >
                    + Add
                  </a>
                </li>
                 
                <hr />
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="grouppage-box"
    >
      <section
        className="grouppage-heading-buttons"
      >
        <section
          className="grouppage-heading"
        >
          <h1>
            Group Name 
          </h1>
          <ul
            className="grouppage-button"
          >
            <li>
              <button
                className="login-default btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Add an Expense
              </button>
               
              <button
                className="Signup-default btn btn-primary"
                data-testid="leavegroup"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Leave Group
              </button>
            </li>
          </ul>
        </section>
        <section
          className="grouppage-center-sec"
        >
          <div
            className="grouppage-center-section-block"
          />
          <div
            className="grouppage-center-section-block"
          >
            <br />
            <div
              className="grouppage-block-border"
            >
              <h2>
                No transactions to display!
              </h2>
            </div>
          </div>
        </section>
      </section>
    </div>
    <div
      className="grouppage-right"
    />
    <div
      className="title"
    >
       
    </div>
    <h5>
      NONE HAVE ACCEPTED THE INVITES TO THE GROUP!
    </h5>
  </div>,
  " ",
]
`;
